### ISSUE TITLE
No surround on Android TV but surround on smartphone

### ISSUE THREAD

**Issue by Rutjan666** [moonlight-stream/moonlight-android#1111]
Hi guys,

Here's an issue I can't wrap my head around: I can't get surround sound from my Android TV with the Moonlight app!

The set up:
- Samsung Home Cinema HT-C5900 receiver
- Samsung TV UE32ES6100
- Connect TV SFR with Android TV Pie 9.7.17

The Connect TV player is provided by my ISP. There's no documentation whatsoever conserning the audio options which are as follows:
- Dolby sounds:
     * OFF
     * LINE (currently set on this)
     * RF
- Audio Mixing:
     * Toggled off, it reads: "support 7.1 channels through Dolby Audio playback"
     * Toggled on, it reads: "support 5.1 channels through Dolby Audio playback" (currently set on this)
- Audio digital format :
     * PCM
     * SPDIF
     * AUTO (currently set on this)
     * Manual (which opens up several toggles):
          - Dolby Atmos in Dolby Digital Plus
          - Dolby Digital
          - Dolby Digital Plus
          - DTS
          - DTS-HD
          - AAC
          - Dolby TrueHD
     * DTS DRC sound scale with options from 0 to 100% going 10 by 10 (currently set to 0)

The Connect TV player HDMI out is plugged to one of the HDMI in of the 5.1 receiver, and the HDMI out of the receiver is plugged to the HDMI ARC of the TV.
Additionally, I have the SPDIF of the Connect TV plugged to the SPDIF of the 5.1 receiver, so I can enable Dolby Pro Logic when the audio source is stereo (in that configuration, I unplug the HDMI from the receiver and put it on the TV directly).

I'm getting 5.1 surround sound on apps such as Prime Video or my Synology Nas with DS Video app, but not from Moonlight which is stuck to plain stereo.

I tried different settings, like putting Dolby sounds to OFF or Audio digital format to PCM (from what I read, Nvidia gamestream only supports PCM audio output) but nothing works, still plain stereo in Moonlight.
If I use SPDIF instead of HDMI, then the sound comes from the rear speakers only!

However, if I use Moonlight on my Samsung Tab S6 and connect it to the 5.1 receiver via a HDMI hub, 5.1 sound works fine.

So it has to be the settings from the Connect TV which prevent it.

Any guesses?

**Comment by cgutman** [comment]
I think that "through Dolby Audio playback" part is the problem. Since GameStream (and Moonlight) can only provide PCM audio, it gets downmixed to Stereo due to bandwidth limitations of ARC. Most other surround media content uses a Dolby codec with compression that can work over ARC.

Sending 5.1 and 7.1 PCM audio through ARC requires all devices have eARC support, but I suspect your device, TV, or 5.1 receiver doesn't support it.

> If I use SPDIF instead of HDMI, then the sound comes from the rear speakers only!

I'm not sure why it only came out of the rear speakers but SPDIF is also limited to stereo according to https://www.startech.com/en-us/faq/uncompressed-digital-spdif-surround-sound-limitati

**Comment by Rutjan666** [comment]
Thanks for your insight.

In my configuration, the Android Connect TV is plugged to the HDMI of the AV receiver through a "standard" HDMI port, not an ARC port.

It's the HDMI out of the receiver that gets into the HDMI ARC port of the TV, but it's just for the image (as well as power and volume control over one remote).

Strangely, when I set audio digital format to PCM on the Android Connect TV, I would have guessed the surround sound would work in Moonlight.

**Comment by cgutman** [comment]
Odd, but it certainly wouldn't be the first time someone had strange issues with an Android TV box outputting PCM.

Do you happen to have any other 5.1/7.1 PCM content (such a test video or audio sample) that you could try?

**Comment by Rutjan666** [comment]
Well, Amazon Prime outputs 5.1 just fine and the videos stored on my Synology NAS (except form proprietary formats like DTS which requires a license Synology no longer provides).

**Comment by cgutman** [comment]
Right, I'm wondering if you could test a sample that we know to be PCM such as https://www.lynxstudio.com/downloads/aes16e/sample-wav-file-zip-encoded-44-1khz-pcm-24-5-1-surround/

**Comment by Rutjan666** [comment]
I tested the audio file as suggested, opening the desktop in Moonlight and as expected, the output is stereo only (even though I select 5.1 in my Windows sound settings).
Of course, if I connect my PC via HDMI directly into the AV receiver the output is 5.1 and the audio file is played correctly through each channel.

**Comment by cgutman** [comment]
oh, I meant playing it directly on your Android TV.

**Comment by Rutjan666** [comment]
Oh my bad! Played the file on VLC and the output is "somewhat" stereo: I'm getting sound from the front left, right and even center, but nothing from the subwoofer or the rear speakers (even though there's a faint sound from those but the line "surround left, surround right" is still played on the front speakers...
I tried shuffling the audio settings of the Android TV but nothing works.

**Comment by cgutman** [comment]
okay, so definitely something with the Android TV device itself not playing PCM 5.1 correctly, rather than a Moonlight-specific bug.

Probably not much we can do here :(

**Comment by Rutjan666** [comment]
Just for the record, I tested the Steam Link app, and it's the same thing, stereo only, so definitely not a Moonlight-only issue!

Thanks for your time anyway.

**Comment by kiam001** [comment]
To fixl it Moonlight Need Convert audio in To Dolby Digital 5.1

**Comment by Rutjan666** [comment]
Problem is, Dolby Digital decoding requires a license that isn't free...

**Comment by kiam001** [comment]
Kodi Client have in settings a Working Transcode funtion

**Comment by kiam001** [comment]
https://forum.kodi.tv/showthread.php?tid=330220

**Comment by Rutjan666** [comment]
What does Kodi have to do with Moonlight?

**Comment by kiam001** [comment]
Both opensource
